x-pack/plugins/apm/public/components/alerting/ui_components/chart_preview/index.tsx

```diff
@@ -64,36 +69,73 @@ export function ChartPreview({
     opacity: thresholdOpacity,
   };
 
+  const DEFAULT_DATE_FORMAT = 'Y-MM-DD HH:mm:ss';
+
+  const tooltipProps: TooltipProps = {
+    headerFormatter: ({ value }) => {
+      const dateFormat =
+        (uiSettings && uiSettings.get(UI_SETTINGS.DATE_FORMAT)) ||
+        DEFAULT_DATE_FORMAT;
+      return moment(value).format(dateFormat);
+    },
+  };
+
+  const barSeries = useMemo(() => {
+    return series.reduce<
+      Array<{ x: number; y: Maybe<number>; groupBy: string | undefined }>
+    >((acc, serie) => {
+      const barPoints = serie.data.reduce<
+        Array<{ x: number; y: Maybe<number>; groupBy: string | undefined }>
+      >((pointAcc, point) => {
+        return [...pointAcc, { ...point, groupBy: serie.name }];
+      }, []);
+      return [...acc, ...barPoints];
+    }, []);
+  }, [series]);
```

Comment:
Not 100% sure if this is the intention, but I think this can be simplified like so:

```suggestion
  const barSeries = useMemo(() => {
    return series.flatMap((serie) =>
      serie.data.map((point) => ({ ...point, groupBy: serie.name }))
    );
  }, [series]);
```

Reply:
You are right. I updated the code.